signer/src/block_observer.rs

```diff
@@ -187,24 +184,41 @@ where
         Ok(())
     }
 
+    /// Fetch deposit requests from Emily and store the validated ones into
+    /// the database.
     #[tracing::instrument(skip(self))]
     async fn load_latest_deposit_requests(&mut self) -> Result<(), Error> {
-        let deposit_requests = self.emily_client.get_deposits().await?;
+        let mut deposit_requests = Vec::new();
+        let mut failed_requests = Vec::new();
 
-        for request in deposit_requests {
+        for request in self.emily_client.get_deposits().await? {
             let deposit = request
                 .validate(&self.context.get_bitcoin_client())
                 .await
                 .inspect_err(|error| tracing::warn!(%error, "could not validate deposit request"));
 
-            if let Ok(Some(deposit)) = deposit {
-                self.deposit_requests
-                    .entry(deposit.info.outpoint.txid)
-                    .or_default()
-                    .push(deposit);
+            match deposit {
+                // Happy path, we've successfully validated the deposit
+                // request.
+                Ok(Some(deposit)) => deposit_requests.push(deposit),
+                // This happens when the transaction has failed the first
+                // step of validation or has passed the first step but we
+                // don't recongnize the x-only public key in the deposit
+                // script.
+                Err(Error::SbtcLib(_)) | Err(Error::UnknownAggregateKey(_, _)) => {
+                    failed_requests.push(request.outpoint);
```

Comment:
I'm not entirely sure what, if anything, we should do with these failed requests. Should we tell Emily, persist them to the database, store them in memory but with a capped size, or do nothing?

Reply:
I think we should report them to Emily so that Emily can stop giving it to us each round.

Reply:
> I think we should report them to Emily so that Emily can stop giving it to us each round. Otherwise we spam the logs every block with the same validation failures.
> 
> Two other plus's to having it to emily are 1) user feedback, 2) analytics.
> 
> But for short-term we could just have an [indexset](https://docs.rs/indexmap/latest/indexmap/set/struct.IndexSet.html) or something and start popping after like 1000 entries.

The report-to-emily change makes a lot of sense, but I was hesitant to do that because one potentially mistaken signer could then effectively block a deposit. I think we need to open a design issue for this in order for us to figure out what to do long term.

I like the short-term change solution, but I think it should be done in a separate PR with it's own ticket describing why it's necessary and so on.

Reply:
Yeah I thought about that multiple signers too... I was thinking that it would actually be best for Emily to store updates from signers in an array, if nothing else than for centralized debugging, but we could also use majority rules for when a deposit is actually rejected.

Reply:
I created https://github.com/stacks-network/sbtc/issues/683 for the signers side, still need to create one for any potential emily work.

Reply:
Hmmm, I'm not sure that we want to do this on the Emily side either. I'll leave this as is for now and we can settle on a long term plan sometime later.